src/core/database/query.helpers.ts

```diff
@@ -995,10 +995,31 @@ export function listWithSecureObject(props: string[]) {
   return props.map((x) => returnWithSecurePropertyClauseForList(x)).join(', ');
 }
 
+export function listWithSecureObjectAsObject(props: string[]) {
+  const result: any = {};
+  props.map((x) => {
+    result[x] = `{
+        value: coalesce(${x}.value, null),
+        canRead: coalesce(${x}ReadPerm.read, false),
+        canEdit: coalesce(${x}EditPerm.edit, false)
+      }
+    `;
+  });
+  return result;
+}
+
 export function addBaseNodeMetaPropsWithClause(props: string[]) {
   return props.map((x) => `${x}: node.${x}`).join(', ');
 }
 
+export function addBaseNodeMetaPropsWithClauseAsObject(props: string[]) {
+  const result: any = {};
+  props.map((x) => {
+    result[x] = `node.${x}`;
+  });
+  return result;
```

Comment:
`.map()` usage is incorrect here because the function has a return type that should be used. If there is not result then a `for..of` loop should be used.

For the case of converting a list to a map/object we have another `mapFromList` helper.
The iteratee should return a tuple of a `[key, value]` that is merged into the resulting object. Like this:
```suggestion
  return mapFromList(props, prop => [prop, `node.${prop}`]);
```